dao/secret_dao.go

```diff
@@ -0,0 +1,121 @@
+package dao
+
+import (
+	"context"
+	"fmt"
+
+	m "github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/util"
+	"gorm.io/gorm"
+)
+
+const SecretResourceType = "Tenant"
```

Comment:
If we change this to only being set in the create function we should make this private.